pkg/controller/astarte/reconcile/utils.go

```diff
@@ -390,11 +390,22 @@ func getImageForClusteredResource(defaultImageName, defaultImageTag string, reso
 	return image
 }
 
-func getDeploymentStrategyForClusteredResource(cr *apiv1alpha1.Astarte, resource apiv1alpha1.AstarteGenericClusteredResource) appsv1.DeploymentStrategy {
-	if resource.DeploymentStrategy != nil {
+func getDeploymentStrategyForClusteredResource(cr *apiv1alpha1.Astarte, resource apiv1alpha1.AstarteGenericClusteredResource, component apiv1alpha1.AstarteComponent) appsv1.DeploymentStrategy {
+
+	switch {
+	case resource.DeploymentStrategy != nil:
 		return *resource.DeploymentStrategy
+	case component == apiv1alpha1.DataUpdaterPlant && resource.DeploymentStrategy == nil:
```

Comment:
The `resource.DeploymentStrategy == nil` is redundant if you place the `cr.Spec.DeploymentStrategy != nil` case above this